### PR TITLE
CLI for Azure load testing

### PR DESCRIPTION
**Pull request by venkatr21**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
**az load**
_az load create
az load update
az load list
az load show
az load delete_

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?
- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Azure load testing

**Comment by venkatr21** [comment]
Bringing up this point, we would like to do the review for load testing CLI currently, and proceed with release around our service GA date (Jan 2023)

**Comment by jsntcy** [comment]
@venkatr21, we have already helped generate the code in https://github.com/Azure/azure-cli-extensions/pull/5514 and we can continue on that work.
BTW, do you mean we should release around the service GA date instead of releasing earlier (early Dec) before GA date?

**Comment by venkatr21** [comment]
@jsntcy Yes, we would like to have CLI to be released around GA date, and not very much earlier.
The PR which you have mentioned #5514 seems to be a skeleton generated from aaz , the current PR(#5558) is almost in a completion state. Can we continue the work/review in this current PR?

**Comment by jsntcy** [comment]
> @jsntcy Yes, we would like to have CLI to be released around GA date, and not very much earlier. The PR which you have mentioned #5514 seems to be a skeleton generated from aaz , the current PR(#5558) is almost in a completion state. Can we continue the work/review in this current PR?

Sure, please continue to work on your PR and we can help review the PR.

**Comment by venkatr21** [comment]
@jsntcy @yanzhudd Thank you, can we please add a do not merge tag to this PR (so that it doesn't get merged until the GA release date of load test service).

**Comment by jsntcy** [comment]
Yes, added the label.

File: src/load/azext_load/aaz/latest/load/_create.py
Comment (kairu-ms) [inline src/load/azext_load/aaz/latest/load/_create.py:25]:
```suggestion
        az load create -n sample-resource -g sample-rg -l westus2 --identity-type SystemAssigned,UserAssigned --user-assigned "{'/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sample-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/sample-mi':{}}"
```

File: src/load/azext_load/aaz/latest/load/_create.py
Comment (kairu-ms) [inline src/load/azext_load/aaz/latest/load/_create.py:26]:
```suggestion
        az load create -n sample-resource -g sample-rg -l westus2 --identity-type SystemAssigned,UserAssigned --user-assigned "{'/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sample-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/sample-mi':{}}" --encryption-key https://sample-kv.vault.azure.net/keys/samplekey/2d1ccd5c50234ea2a0858fe148b69cde --encryption-identity /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sample-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/sample-mi
```

File: src/load/azext_load/aaz/latest/load/_create.py
Comment (kairu-ms) [inline src/load/azext_load/aaz/latest/load/_create.py:21]: Recommend separating the following 5 commands in to 5 examples.

File: src/load/azext_load/aaz/latest/load/_delete.py
Comment (kairu-ms) [inline src/load/azext_load/aaz/latest/load/_delete.py:23]: recommend separating the following two commands into 2 examples

File: src/load/azext_load/aaz/latest/load/_update.py
Comment (kairu-ms) [inline src/load/azext_load/aaz/latest/load/_update.py:24]:
```suggestion
        az load update --name sample-resource --resource-group sample-rg --identity-type SystemAssigned,UserAssigned --user-assigned "{'/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sample-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/sample-mi':{}}" --encryption-key https://sample-kv.vault.azure.net/keys/samplekey2/2d1ccd5c50234ea2a0858fe148b69cde --encryption-identity SystemAssigned
```

File: src/load/azext_load/operations/create.py
Comment (kairu-ms) [inline src/load/azext_load/operations/create.py:32]:
```suggestion
            encryption_identity_id = args.encryption_identity.to_serialized_data()
```

File: src/load/azext_load/operations/create.py
Comment (kairu-ms) [inline src/load/azext_load/operations/create.py:29]:
```suggestion
        identity_type_str = args.identity_type.to_serialized_data()
```

File: src/load/azext_load/operations/update.py
Comment (kairu-ms) [inline src/load/azext_load/operations/update.py:29]:
```suggestion
        identity_type_str = args.identity_type.to_serialized_data()
```

File: src/load/azext_load/operations/update.py
Comment (kairu-ms) [inline src/load/azext_load/operations/update.py:32]:
```suggestion
            encryption_identity_id = args.encryption_identity.to_serialized_data()
```

File: src/load/azext_load/operations/update.py
Comment (kairu-ms) [reply src/load/azext_load/operations/update.py:29]: When you use .to_serialized_data() of AAZStringArg, the return value type will be string, None (if argument is nullable and user pass `null`) or AAZUndefined (not has value).

**Comment by kairu-ms** [comment]
Please update .github/CODEOWNERS file to fix CI issues.

File: src/load/azext_load/operations/update.py
Comment (venkatr21) [reply src/load/azext_load/operations/update.py]: Updated it, as recommended

File: src/load/azext_load/operations/update.py
Comment (venkatr21) [reply src/load/azext_load/operations/update.py]: Updated it, as recommended

File: src/load/azext_load/aaz/latest/load/_delete.py
Comment (venkatr21) [reply src/load/azext_load/aaz/latest/load/_delete.py]: Updated it, as two examples

**Comment by venkatr21** [comment]
I am attaching the PR of AAZ repo, where we are updating the models generated by the aaz tool.
https://github.com/Azure/aaz/pull/48

**Comment by jsntcy** [comment]
@venkatr21, If the PR is ready for review, please remove "draft" status of the PR.

**Comment by venkatr21** [comment]
@jsntcy It is ready for the review as the dev work is completed from our end. But as discussed, we would like to merge and release the PR around our GA date (mid of Jan 2023)

**Review by kairu-ms — APPROVED** [review]
LGTM

**Pull request merged**
